### ISSUE TITLE
[Bug]: smartd logging broken after upgrade to v1.46

### ISSUE THREAD

**Issue by kYwzor** [netdata/netdata#17989]
### Bug description

I used to be able to track S.M.A.R.T. data on Netdata, but after upgrading from v1.45.6 to v1.46.0, the charts no longer have any data. An example of a broken chart can be seen here: ![example](https://github.com/netdata/netdata/assets/19508557/d58b6a70-928d-4162-9d1c-83a63d970b61)

The break-point occurs at the time of upgrade. There doesn't seem to be any new/alternative charts that contain this data. This may be related to the changes to the collector from Python to Go (see #17536). Upgrading from v1.46.0 to v1.46.1 did not fix the issue.

### Expected behavior

The expectation was to upgrade Netdata and not lose any configured logging, including S.M.A.R.T. data.

### Steps to reproduce

1. Set up the smartctl collector on v1.45.6 
2. Confirm that the "smartd log" charts have data.
3. Upgrade to v1.46.0. 
4. Notice that the SMART data isn't being logged anymore.

### Installation method

kickstart.sh

### System info

```shell
Linux ***** 6.1.0-21-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.90-1 (2024-05-03) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="12"
/etc/os-release:VERSION="12 (bookworm)"
/etc/os-release:VERSION_CODENAME=bookworm
/etc/os-release:ID=debian
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.46.1
    Installation Type __________________________________________ : binpkg-deb
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ : debian 12
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /usr/share/netdata/web
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 6.1.0-21-amd64
    Operating System ___________________________________________ : Debian GNU/Linux
    Operating System ID ________________________________________ : debian
    Operating System ID Like ___________________________________ : unknown
    Operating System Version ___________________________________ : 12 (bookworm)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 36
    CPU Frequency ______________________________________________ : 3000000000
    RAM Bytes __________________________________________________ : 540509007872
    Disk Capacity ______________________________________________ : 34030667317248
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : YES
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : YES
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : NO
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : YES
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

**Comment by ilyam8** [comment]
Hi. `smartd_log` (python version) was deprecated in favor of [go.d/smartctl](https://github.com/netdata/netdata/tree/master/src/go/collectors/go.d.plugin/modules/smartctl#readme), see [v1.46.0 deprecation notice](https://github.com/netdata/netdata/releases/tag/v1.46.0#v1460-deprecation-notice).

- New installations don't have `smartd_log`.
- Existing installations (upgrade, your case) still have it. If you need it for some reason - you can still enable it in `python.d.conf`.

**Comment by kYwzor** [comment]
@ilyam8 thanks for the clarification. My issue is that no alternative seems to have been enabled, so I currently have no SMART logging regardless of collector. Do I need to somehow manually activate go.d/smartctl? The docs seem to imply it will be active as long as the necessary setup is in place (which it is).

**Comment by ilyam8** [comment]
> If you need it for some reason - you can still enable it in python.d.conf.

I was wrong, it is not disabled by default in [python.d.conf](https://github.com/netdata/netdata/blob/master/src/collectors/python.d.plugin/python.d.conf). So it works for existing installations (upgrade). If you have problems with it, you can still debug them like any other python.d module using the debug mode.

> Do I need to somehow manually activate go.d/smartctl?

No. It should work if the [prerequisites](https://learn.netdata.cloud/docs/collecting-metrics/hardware-devices-and-sensors/s.m.a.r.t.#prerequisites) are met.

---

To troubleshoot issues: [debug mode](https://learn.netdata.cloud/docs/collecting-metrics/hardware-devices-and-sensors/s.m.a.r.t.#debug-mode).

Where is your `smartctl` located? Should be in `/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin`. If you have it in a different directory, create a symlink to (for example) `usr/local/bin` and restart Netdata.

**Comment by kYwzor** [comment]
`smartctl` is at `/usr/sbin/smartctl`, so that wasn't the issue.

I figured it out: I had an empty file at go.d/smartctl.conf which apparently causes the collector to not do anything. After removing it the logging started working correctly. I am a bit perplexed though, because I do not recall creating this file. The upgrade was done automatically via unattended-upgrades about 2 days ago, and I didn't even notice the missing SMART stuff until today. Looking back at the logs of the past two days, SMART wasn't being logged anywhere until I fixed it now.

Well, regardless, thanks for your work on the new collector!

**Comment by ilyam8** [comment]
> an empty file at go.d/smartctl.conf

That is strange! The [stock configuration file](https://github.com/netdata/netdata/blob/master/src/go/collectors/go.d.plugin/config/go.d/smartctl.conf) is not empty.

Can you please check `python.d/smartd_log.conf`? The fact that python.d/smartd_log stopped working for you is also strange.

**Comment by ilyam8** [comment]
> an empty file at go.d/smartctl.conf which apparently causes the collector to not do anything

Yes, that is how it works. If the file is empty, no jobs will be created. This is why a stock configuration file typically includes at least one job by default.

**Comment by kYwzor** [comment]
Frankly, I may have created the empty file today by mistake while trying to debug the issue, but that still wouldn't explain why nothing was logging right after the upgrade. This is a bit of a head-scratcher. I also manually set `smartctl: yes` on `go.d.conf` while trying to fix it, which shouldn't make a difference as that should be the default, but who knows.

I just confirmed I do not have anything inside `python.d/`.
